### PR TITLE
add requirements

### PR DESCRIPTION
**Pull request by KDanil2000**

### CODE REVIEW

**Comment by KDanil2000** [comment]
1

File: pom.xml [hunk 1]
```diff
@@ -1,56 +1,96 @@
...
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
 
-</project>
+</project>
```
Comment (liliia-ponomarenko) [inline pom.xml]: Let’s  add one empty line

File: pom.xml [hunk 2]
```diff
@@ -1,56 +1,96 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-dbcp2</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (liliia-ponomarenko) [inline pom.xml]: Do you really need all these dependencies?

File: src/main/java/bookstore/OnlineBookStoreApplication.java [hunk 3]
```diff
@@ -1,13 +1,34 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import service.BookService;
 
 @SpringBootApplication
+@ComponentScan(basePackages = {"repository", "service", "bookstore"})
+@EntityScan(basePackages = "model")
```
Comment (liliia-ponomarenko) [inline src/main/java/bookstore/OnlineBookStoreApplication.java:15]: you can remove it

File: src/main/java/exceptions/EntityException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package exceptions;
+
+public class EntityException extends RuntimeException {
```
Comment (liliia-ponomarenko) [inline src/main/java/exceptions/EntityException.java]:
```suggestion
public class EntityNotFoundException extends RuntimeException {
```

File: src/main/java/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (liliia-ponomarenko) [inline src/main/java/model/Book.java:14]: Lombok is a great library but be [attentive when using it with Entity](https://jpa-buddy.com/blog/lombok-and-jpa-what-may-go-wrong/) JFYI

File: src/main/java/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.List;
+import model.Book;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import repository.BookRepository;
+import service.BookService;
+
+@Service
+public class BookServiceImpl implements BookService {
+
```
Comment (liliia-ponomarenko) [inline src/main/java/service/impl/BookServiceImpl.java]:
```suggestion
```

File: src/main/java/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    private final BookRepository bookRepository;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/service/impl/BookServiceImpl.java]: This annotation is redundant (more details in spring documentation)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix a few mistakes ;)

File: src/main/java/bookstore/OnlineBookStoreApplication.java (hunk 3 above)
Comment (Elena-Bruyako) [reply src/main/java/bookstore/OnlineBookStoreApplication.java:15]: Redundant annotations

**Review by Elena-Bruyako — APPROVED** [review]
Good job! Let's remove redundant annotations)

**Pull request merged**